Week-4/assigmnent4.1/random.js

```diff
@@ -0,0 +1,65 @@
+class PromiseSimple {
+    constructor(executionFunction) {
+      this.promiseChain = [];
+      this.handleError = () => {};
+  
+      this.onResolve = this.onResolve.bind(this);
+      this.onReject = this.onReject.bind(this);
+  
+      executionFunction(this.onResolve, this.onReject);
+    }
+  
+    then(handleSuccess) {
+      this.promiseChain.push(handleSuccess);
+  
+      return this;
+    }
+  
```

Comment:
implement finally

Reply:
finally implemented in the promise as well.